### DISCUSSION TITLE
MacOS issues

### DISCUSSION THREAD

**Discussion by nofue** [sqlpage/SQLPage#664] [General]
My MacBook Pro is limited to MacOS 11.x and I cannot see a compelling reason to buy a new one. However, it would be nice to run SQLpage on the MBP "on the go" still. Right now SQLpage runs on my Debian LAN Server, so once I leave the office, I can't work with SQLpage any more. 

Compiling stuff using homebrew doesn't work, as MacOS 11.x isn't supported any more. 
I found 'sqlpage-macos.tgz' but get the error 
`zsh: bad CPU type in executable: ./sqlpage.bin`
which seems to indicate it's compiled for a more modern MacOS. 

Long story short: Is it possible to compile for an older version of MacOS?

**Comment by lovasoa** [comment]
Yes, that should be possible !

You should install rust on your computer: https://www.rust-lang.org/tools/install

Then download the source code for sqlpage: https://github.com/sqlpage/SQLPage/archive/refs/heads/main.zip

Then from the terminal, inside the downloaded source, run: `cargo build --release`

You will then find the sqlpage executable in `target/release/sqlpage`

**Reply by nofue** [reply]
Worked like a charm. Well, a rusty charm, pun intended. I got plenty of complaints about files which couldn't be downloaded -- probably an issue with MacOS and safety. Anyway, copying each of the proposed curl <URL> lines and running it as the local user, then restarting the build eventually got me there. The rest was a piece of pie. Thanks for your assistance.
